service/src/main/java/com/codeforcommunity/requester/MapRequester.java

```diff
@@ -41,85 +46,132 @@ public MapRequester(Vertx vertx) {
    * Makes a request to the ArcGIS mapping service to change the status of the all of the blocks in
    * the given list to the specified new block status.
    *
-   * @param streetIds a list of block IDS.
+   * @param blockIds a list of block IDs.
    * @param updateTo The block status all blocks should be updated to.
    */
-  public void updateStreets(List<String> streetIds, BlockStatus updateTo) {
-    updateLayers(streetIds, updateTo, this.tokenFuture);
+  public void updateBlocks(List<String> blockIds, BlockStatus updateTo) {
+    updateLayers(blockIds, updateTo, this.tokenFuture);
   }
 
   /** Make a request to update the ArcGIS feature layer */
   private Future<Void> updateLayers(
-      List<String> streetFids, BlockStatus updateTo, Future<String> tokenFuture) {
+      List<String> blockIds, BlockStatus updateTo, Future<String> tokenFuture) {
     logger.info("Making request to update blocks to " + updateTo.name());
 
-    JsonArray updateJson = new JsonArray();
-    streetFids.forEach(
-        (id) -> {
-          updateJson.add(JsonObject.mapFrom(new MapRequest(id, updateTo)));
-        });
-
-    Future<Void> updateFuture =
+    Future<JsonArray> getBlockFIDsFuture =
         tokenFuture.compose(
             tokenString -> {
               MultiMap formData =
                   MultiMap.caseInsensitiveMultiMap()
                       .add("f", "json")
                       .add("token", tokenString)
-                      .add("updates", updateJson.encode());
+                      .add("where", "ID IN (" + String.join(",", blockIds) + ")")
+                      .add("outFields", "FID");
 
               return Future.future(
                   promise -> {
                     client
-                        .postAbs(featureLayerRoute)
+                        .postAbs(featureLayerQueryRoute)
                         .putHeader("content-type", "multipart/form-data")
                         .sendForm(
                             formData,
-                            (ar) -> {
-                              if (ar.succeeded()) {
-                                HttpResponse<Buffer> httpResponse = ar.result();
+                            (asyncResult -> {
+                              if (asyncResult.succeeded()) {
+                                HttpResponse<Buffer> httpResponse = asyncResult.result();
 
                                 if (httpResponse.statusCode() == 200) {
                                   JsonObject responseBody = httpResponse.bodyAsJsonObject();
-                                  if (responseBody.containsKey("error")
```

Comment:
This logic block checking if the token has expired has to be present on both API calls. Move this checking logic into a helper function so that we can use it in both futures

Reply:
????

Reply:
I thought you were referring to the + features, not the error oops. I get what you're saying now.